src/services/container.js

```diff
@@ -1,43 +1,38 @@
 //@flow
 
-import db from 'BITNATION-Panthalassa/src/database/db'
-import secureStorage from './panthalassa/secureStorage'
-import ethUtils from 'BITNATION-Panthalassa/src/ethereum/utils'
-import web3 from 'BITNATION-Panthalassa/src/ethereum/web3'
-import osDeps from './panthalassa/osDependencies'
-import ethDaemon from './panthalassa/ethDaemon'
+import db from 'BITNATION-Panthalassa/src/database/db';
+import secureStorage from './panthalassa/secureStorage';
+import ethUtils from 'BITNATION-Panthalassa/src/ethereum/utils';
+import web3 from 'BITNATION-Panthalassa/src/ethereum/web3';
+import osDeps from './panthalassa/osDependencies';
+import ethDaemon from './panthalassa/ethDaemon';
 import wallet from 'BITNATION-Panthalassa/src/ethereum/wallet';
 import profile from 'BITNATION-Panthalassa/src/profile/profile';
 
-const EventEmitter = require('eventemitter3');
+async function createContainer() {
+  const EventEmitter = require('eventemitter3');
 
-const DB_PATH = 'panthalassa';
+  const DB_PATH = 'panthalassa';
 
-const ee = new EventEmitter();
-const dbInstance = db(DB_PATH);
-const ethUtilsInstance = ethUtils(secureStorage, ee, osDeps);
-const ethWeb3Instance = web3(ethDaemon, ee, ethUtilsInstance);
-const ethWallet = wallet(ethUtilsInstance, ethWeb3Instance, dbInstance);
-const profileInstance = profile(dbInstance, ethUtilsInstance);
+  const ee = new EventEmitter();
+  const dbInstance = db(DB_PATH);
+  const ethUtilsInstance = ethUtils(secureStorage, ee, osDeps);
+  const ethWeb3Instance = await web3(ethDaemon, ee, ethUtilsInstance);
+  const ethWallet = wallet(ethUtilsInstance, ethWeb3Instance, dbInstance);
+  const profileInstance = profile(dbInstance, ethUtilsInstance);
 
-/**
- * @alias src/services/container.js
- * @desc An object holding all available services
- * @type {{eventEmitter, panthalassa: {database: DB, ethereum: {utils: EthUtilsInterface, web3: function(), wallet: WalletInterface}, profile: Profile}}}
- */
-const container = {
+  return {
     eventEmitter: ee,
     panthalassa: {
-        database: dbInstance,
-        ethereum: {
-            utils: ethUtilsInstance,
-            web3: ethWeb3Instance,
-            wallet: ethWallet
-        },
-        profile: profileInstance
-    }
-};
+      database: dbInstance,
+      ethereum: {
+        utils: ethUtilsInstance,
+        web3: ethWeb3Instance,
+        wallet: ethWallet,
+      },
+      profile: profileInstance,
+    },
+  };
+}
 
-Object.freeze(container);
-
-export default container;
+export default createContainer();
```

Comment:
But isn't a container created again and again? If I call the exported function?

Reply:
No, we export not the function, but the result of function, which is a promise. So, it will be called only once at start. I have checked that by putting logs into createContainer function.

Reply:
Ah yeah, I just see that you are calling it xD Sry for asking.